### DISCUSSION TITLE
New XSharedPreferences not work

### DISCUSSION THREAD

**Discussion by MichaelZhuravsky** [LSPosed/LSPosed#1079] [Q&A]
Hello

I've created a test module to check how new-prefs work. Unfortunatelly, I could not make it work as required. 

Android 11, MIUI 12.5.3 stable

Test module source: https://MZhuravsky@bitbucket.org/MZhuravsky/testmodule.git
Test module signed apk: https://bitbucket.org/MZhuravsky/testmodule/downloads/app-release.apk

I've added xposedsharedprefs as meta-data + used Context.MODE_WORLD_READABLE.

As I can see in LoadedApkGetCLHooker.hookNewXSP method adding meta-data is enought, but there is no proper result whatever I do.

So, what else should I do to make prefs located in 'data/misc' folder?

**Comment by ghost** [comment]
I don't rember the code, but I think you should set xposedminversion to 93 and not 30+

EDIT
https://github.com/LSPosed/LSPosed/blob/b98979d69b537d9a4907df736fe5fae882d4f448/core/src/main/java/de/robv/android/xposed/XSharedPreferences.java#L171-L184
My fault with xposedsharedprefs should work 

EDIT2
[This](https://bitbucket.org/MZhuravsky/testmodule/src/0e0f7f7c5d035b67fda7ea14701464c72a55da8f/app/src/main/java/ru/mike/testmodule/XposedMod.kt#lines-28) in your code is wrong.
In xposed use the file BuildConfig.APPLICATION_ID + "_preferences" and in MainActivity pref_file

**Reply by ghost** [reply]
I tested your module and write correctly in /data/misc/lspd_xxx/prefs/ru.mike.testmodule the file pref_file.xml after you enable it manager.

**Comment by MichaelZhuravsky** [comment]
Thx for your help!

1. Don't understand about pref names. There is no requirements for new-pres as I can see
According to the Wiki I have to write (in MainActivity) via prefs
`getSharedPreferences("pref_file", Context.MODE_WORLD_READABLE)`
and read (in the module) via prefs
`XSharedPreferences(BuildConfig.APPLICATION_ID, "pref_file")`
Is it wrong? Where I should use BuildConfig.APPLICATION_ID + "_preferences"? Is it just code-style namig?

**Comment by MichaelZhuravsky** [comment]
2. Magic! Now I've recompiled the module without minifyEnabled and it works! Does the new-prefs depends on minifyEnabled? Is there proper rules for proguard?

Update: I have no idea about reasons, but today it works with proguard enabled too.

**Comment by MichaelZhuravsky** [comment]
I've got the bug: LSPosed Manager causes it. Version 1.5.3 (5984)

To reproduce it download the test app for the module (module's scope) above: https://bitbucket.org/MZhuravsky/testmodule/downloads/test-app-release.apk

Now after install just enable the module in the Manager, the test app will be already checked by the scope.
Nothing more, do NOT uncheck/check recommended app and any others. The new prefs will not be applied.

Uninstall the module and install it again. Now enable it, and uncheck & check again the recommended app (or any other). Now the prefs work as expected

**Reply by nift4** [reply]
I still have this problem: module hook from LSPosed is not hooking unless you manually tick one (any) app

**Reply by nift4** [reply]
(I am talking about this: ```When enabled, LSPosed will automatically hook ContextImpl.getPreferencesDir() so that your Context.getSharedPreferences and PreferenceFragment could work without further modification. Also, LSPosed hooks ContextImpl.checkMode(int) so that you can easily get a world readable shared preference by passing Context.MODE_WORLD_READABLE to Context.getSharedPreferences.```)
